src/components/quack.js

```diff
@@ -1,24 +1,38 @@
 import { SOUND_QUACK, SOUND_SPECIAL_QUACK } from "../systems/sound-effects-system";
 
+const INTERACTORS = ["rightHand", "leftHand", "rightRemote", "leftRemote"];
+
 AFRAME.registerComponent("quack", {
   schema: {
     quackPercentage: { default: 1 },
     specialQuackPercentage: { default: 0.01 }
   },
 
   init: function() {
-    this._handleGrabStart = this._handleGrabStart.bind(this);
+    this.wasInteracting = false;
+    NAF.utils.getNetworkedEntity(this.el).then(networkedEntity => {
+      this.networkedEntity = networkedEntity;
+    });
   },
 
-  play: function() {
-    this.el.object3D.addEventListener("interact", this._handleGrabStart);
-  },
+  tick: function() {
+    const interaction = AFRAME.scenes[0].systems.interaction;
+    const entity = this.networkedEntity || this.el;
+    let isInteracting = false;
+    for (let i = 0; i < INTERACTORS.length; i++) {
+      if (interaction.state[INTERACTORS[i]].held === entity) {
+        isInteracting = true;
+        if (!this.wasInteracting) {
+          this.quack();
+        }
+        break;
+      }
+    }
```

Comment:
There's an `isHeld` helper you could use:
```
const isInteracting = interaction.isHeld(entity);
if (isInteracting && !this.wasInteracting) {
  this.quack();
}
this.wasInteracting = isInteracting;
```

Reply:
ah yea that's much better, thanks